### PR TITLE
Add modvar and create new pager

### PR DESCRIPTION
**Pull request by craigh**
add modvar for items per page to ZAuth mappings list

also remove legacy pager and user doctrine pager with custom wrapper

| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | -
| Refs tickets      | refs #2915
| License           | MIT
| Changelog updated | yes

### CODE REVIEW

**Comment by Guite** [comment]
Did you also look at https://github.com/KnpLabs/KnpPaginatorBundle ?

File: src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{% if paginator.route is empty and app.request.attributes.has('_route') %}
+    {% do paginator.setRoute(app.request.attributes.get('_route')) %}
+{% endif %}
+{% if paginator.hasToPaginate %}
+    <nav aria-label="Page navigation">
```
Comment (Guite) [inline src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig]: Please add `{% trans %}`

File: src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        <ul class="pagination justify-content-center">
+            {% if paginator.hasPreviousPage %}
+                {% do paginator.setRouteParameter('page', paginator.previousPage) %}
+                <li class="page-item">
+                    <a class="page-link" href="{{ path(paginator.route, paginator.routeParameters) }}">
```
Comment (Guite) [inline src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig]: Please add a link title.

File: src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                        <i class="fa fw fa-arrow-left"></i>
+                    </a>
+                </li>
+            {% else %}
+                <li class="page-item disabled">
+                    <a class="page-link" href="#" aria-label="Previous">
```
Comment (Guite) [inline src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig]: Please add `{% trans %}` and a link title.

File: src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                    <li class="page-item">
+                        <a class="page-link" href="{{ path(paginator.route, paginator.routeParameters) }}">{{ i }}</a>
+                    </li>
+                {% endif %}
+            {% endfor %}
+
+            {% if paginator.hasNextPage %}
+                {% do paginator.setRouteParameter('page', paginator.nextPage) %}
+                <li class="page-item">
+                    <a class="page-link" href="{{ path(paginator.route, paginator.routeParameters) }}">
```
Comment (Guite) [inline src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig]: Please add a link title.

File: src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                        <i class="fa fw fa-arrow-right"></i>
+                    </a>
+                </li>
+            {% else %}
+                <li class="page-item disabled">
+                    <a class="page-link" href="#" aria-label="Previous">
```
Comment (Guite) [inline src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig]: Please add `{% trans %}` and a link title. Also replace `Previous` by `Next`.

File: src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                        <span aria-hidden="true"><i class="fa fw fa-arrow-right"></i></span>
+                    </a>
+                </li>
+            {% endif %}
```
Comment (Guite) [inline src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig]: Would be great to have "last page", too.

File: src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig [hunk 7]
```diff
@@ -0,0 +1,51 @@
+{% if paginator.route is empty and app.request.attributes.has('_route') %}
+    {% do paginator.setRoute(app.request.attributes.get('_route')) %}
+{% endif %}
+{% if paginator.hasToPaginate %}
+    <nav aria-label="Page navigation">
+        <ul class="pagination justify-content-center">
+            {% if paginator.hasPreviousPage %}
```
Comment (Guite) [inline src/Zikula/CoreBundle/Resources/views/Paginator/Paginator.html.twig]: Would be great to have "first page", too.

**Comment by Guite** [comment]
> Did you also look at https://github.com/KnpLabs/KnpPaginatorBundle ?

Does your "thumbs down" mean that you didn't look or that you don't like it?

**Comment by Guite** [comment]
How about deprecating the old pagers?

**Comment by craigh** [comment]
I didn't look at it until your link. I don't like it because Doctrine already provides most of the solution. This solution is simple and doesn't require a full-blown bundle.

**Pull request merged**